restli-tools/src/main/java/com/linkedin/restli/tools/snapshot/gen/PegasusSchemaSnapshotExporter.java

```diff
@@ -0,0 +1,130 @@
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package com.linkedin.restli.tools.snapshot.gen;
+
+import com.linkedin.data.schema.AbstractSchemaEncoder;
+import com.linkedin.data.schema.AbstractSchemaParser;
+import com.linkedin.data.schema.DataSchema;
+import com.linkedin.data.schema.DataSchemaResolver;
+import com.linkedin.data.schema.NamedDataSchema;
+import com.linkedin.data.schema.PegasusSchemaParser;
+import com.linkedin.data.schema.SchemaToPdlEncoder;
+import com.linkedin.data.schema.resolver.MultiFormatDataSchemaResolver;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import org.apache.commons.io.FileUtils;
+
+
+/**
+ * PegasusSchemaSnapshotExporter, generating pegasus schema snapshot(.pdl) files
+ *
+ * @author Yingjie Bi
+ */
+public class PegasusSchemaSnapshotExporter
+{
+  private static final String PEGASUS_SCHEMA_SNAPSHOT_SUFFIX = ".pdl";
+
+  private static final String PDL = "pdl";
+
+  private static final String PDSC = "pdsc";
+
+  /**
+   * Generate pegasus schema snapshot(pegasusSchemaSnapshot.pdl) files to the provided output directory
+   * based on the given input pegasus schemas.
+   *
+   * @param resolverPath schema resolver path
+   * @param inputPath input files directory
+   * @param outputDir output files directory
+   * @return GeneratorResult
+   * @throws IOException
+   */
+  public void export(String resolverPath, String inputPath, File outputDir) throws IOException
+  {
+    List<DataSchema> dataSchemas = parseDataSchema(resolverPath, inputPath);
+    for (DataSchema dataSchema : dataSchemas)
+    {
+      writeSnapshotFile(outputDir, ((NamedDataSchema) dataSchema).getFullName(), dataSchema);
+    }
+  }
+
+  private static List<DataSchema> parseDataSchema(String resolverPath, String inputPath)
+      throws RuntimeException, FileNotFoundException
+  {
+    Iterator<File> iterator = FileUtils.iterateFiles(new File(inputPath), new String[]{PDL, PDSC}, true);
+    List<DataSchema> schemas = new ArrayList<>();
+    while (iterator.hasNext())
+    {
+      DataSchemaResolver resolver = MultiFormatDataSchemaResolver.withBuiltinFormats(resolverPath);
+      File inputFile = iterator.next();
+      String fileExtension = getFileExtension(inputFile.getName());
+      PegasusSchemaParser parser = AbstractSchemaParser.parserForFileExtension(fileExtension, resolver);
+      parser.parse(new FileInputStream(inputFile));
+
+      if (parser.hasError())
+      {
+        throw new RuntimeException(parser.errorMessage());
+      }
```

Comment:
Can you double check if the parser's error message includes the name of the file? (If it does not, let's either update the parser to include it or include it here.)

Reply:
included the name of the file here.